tools/cldr-code/src/main/java/org/unicode/cldr/util/StandardCodes.java

```diff
@@ -553,117 +521,18 @@ public Level getTargetCoverageLevel(String localeId) {
-                if (status == Level.UNDETERMINED) {
-                    System.out.println("Warning: Level unknown on: " + line);
-                }
-                Map<String, Level> locale_status = platform_locale_level.get(organization);
-                if (locale_status == null) {
-                    platform_locale_level.put(organization, locale_status = new TreeMap<>());
-                }
-                locale_status.put(locale, status);
-                if (!locale.equals(ALL_LOCALES)) {
-                    String scriptLoc = parser.getLanguageScript();
-                    if (locale_status.get(scriptLoc) == null)
-                        locale_status.put(scriptLoc, status);
-                    String lang = parser.getLanguage();
-                    if (locale_status.get(lang) == null)
-                        locale_status.put(lang, status);
-                }
-            }
-        } catch (IOException e) {
-            throw new ICUUncheckedIOException("Internal Error", e);
-        }
-
-        // now reset the parent to be the max of the children
-        for (Organization platform : platform_locale_level.keySet()) {
-            Map<String, Level> locale_level = platform_locale_level.get(platform);
-            for (String locale : locale_level.keySet()) {
-                parser.set(locale);
-                Level childLevel = locale_level.get(locale);
-
-                String language = parser.getLanguage();
-                if (!language.equals(locale)) {
-                    Level languageLevel = locale_level.get(language);
-                    if (languageLevel == null || languageLevel.compareTo(childLevel) < 0) {
-                        locale_level.put(language, childLevel);
-                    }
-                }
-                String oldLanguage = language;
-                language = parser.getLanguageScript();
-                if (!language.equals(oldLanguage)) {
-                    Level languageLevel = locale_level.get(language);
-                    if (languageLevel == null || languageLevel.compareTo(childLevel) < 0) {
-                        locale_level.put(language, childLevel);
-                    }
-                }
-            }
-        }
-        // backwards compat hack
-        platform_locale_levelString = new TreeMap<>();
-        platform_level_locale = new EnumMap<>(Organization.class);
-        for (Organization platform : platform_locale_level.keySet()) {
-            Map<String, String> locale_levelString = new TreeMap<>();
-            platform_locale_levelString.put(platform.toString(), locale_levelString);
-            Map<String, Level> locale_level = platform_locale_level.get(platform);
-            for (String locale : locale_level.keySet()) {
-                locale_levelString.put(locale, locale_level.get(locale).toString());
+    private LocalesTxtReader loadPlatformLocaleStatus() {
+        if (localesTxt == null) {
```

Comment:
This can fail in a multi-threading situation. The more reliable and cleaner way do do a lazy load is to have LocalesTxtReader method that returns a private static SINGLETON.

Reply:
I'm not sure this specific one would fail, it might load multiple times.  In any event, StandardCodes is also needed for parsing Locales.txt (circular dependency).  I've fixed this so that the singleton uses StandardCodes.make() to read Locales.txt

Reply:
Double-check loading is notoriously tricky.

Easiest and safest paradigm to use is the https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom

```
public class Something {
    private Something() {}

    private static class LazyHolder {
        static final Something INSTANCE = new Something();
    }

    public static Something getInstance() {
        return LazyHolder.INSTANCE;
    }
}
```

(For more on why it is tricky, you can later read https://en.wikipedia.org/wiki/Double-checked_locking.)

Reply:
that's what is in the PR now